src/generic_ui/polymer/tip.ts

```diff
@@ -0,0 +1,64 @@
+Polymer({
+  close: function() {
+    this.fire('closed');
+    this.remove();
+  },
+  doReposition: function() {
+    var prev = this.previousSibling;
+    if (prev.offsetLeft === 0 && prev.offsetTop === 0 && prev.offsetWidth === 0
+        && prev.offsetHeight === 0) {
+      /* if the DOM is not set up yet, keep checking every half second */
```

Comment:
Can you update the comment then to explain in a bit more detail, and also add something like "TODO: remove setTimeout once we dynamically add/remove tooltips from the DOM rather than use the hidden attribute"?  Then I think this is fine